mindsdb/integrations/handlers/footballApi_handler/footballApi_handler.py

```diff
@@ -0,0 +1,186 @@
+from mindsdb.integrations.libs.response import (
+    HandlerStatusResponse as StatusResponse, HandlerResponse,
+)
+from mindsdb.utilities import log
+from footballAPIClient import footballAPI, FootballAPI
+
+
+class FootballApiHandler(APIHandler):
+    """The Football API handler implementation"""
+
+    def __init__(self, name: str = None, **kwargs):
+        """Initialize the Football API handler.
+
+                Parameters
+                ----------
+                name : str
+                    name of a handler instance
+                """
+        super().__init__(name)
+        self.api_key = None
+        self.account_type = None
+
+        connection_data = kwargs.get('connection_data', {})
+        if 'api_key' in connection_data:
+            self.api_key = connection_data['api_key']
+        elif 'API_KEY' in os.environ:
+            self.api_key = os.environ['API_KEY']
+        if 'account_type' in connection_data:
+            self.account_type = connection_data['account_type']
+
+        self.is_connected = False
+        self.connection_data = connection_data
+        self.connection = None
+        football_api_players = FootballApiTable(self)
+        self._register_table("get_players", football_api_players)
+
+    def connect(self) -> FootballAPI:
+        """Set up the connection required by the handler.
+
+                Returns
+                -------
+                StatusResponse
+                    connection object
+                """
+
+        if self.is_connected is True:
+            return self.connection
+
+        football_client = footballAPI.FootballAPI(self.account_type, self.api_key)
+        self.connection = football_client
+        self.is_connected = True
+        return self.connection
+
+    def check_connection(self) -> StatusResponse:
+        """Check connection to the handler.
+
+                Returns
+                -------
+                StatusResponse
+                    Status confirmation
+                """
+
+        response = StatusResponse(False)
+
+        try:
+            client = self.connect()
+            client.get_status()
+            response.success = True
+        except Exception as e:
+            log.logger.error(f"Error Connecting to Football API: {e}")
+            response.error_message = e
+
+        self.is_connected = response.success
+        return response
+
+    def native_query(self, query: str = None) -> HandlerResponse:
+        """Receive and process a raw query.
+
+                Parameters
+                ----------
+                query : str
+                    query in a native format
+
+                Returns
+                -------
+                HandlerResponse
+                    Request status
+                """
+
+        ast = parse_sql(query, dialect='mindsdb')
+        return self.query(ast)
+
+    def call_football_api(self, method_name: str, **player_params) -> pd.DataFrame:
```

Comment:
If we do create a `PlayersTable` as I have suggested above, I think we should probably move these functions that seem to be specific to extracting player data, to that class. The `call_football_api()` function does not seem to be generic per se, since it takes in the name of a method and then just calls it. I would think that it'd be much simple to call the relevant method inside of each class that we create: `PlayersTable`, `TeamsTable` etc.